src/lib/swapper/LifiSwapper/LifiSwapper.ts

```diff
@@ -0,0 +1,137 @@
+import { filterAssetIdsBySellable } from 'lib/swapper/LifiSwapper/filterAssetIdsBySellable/filterAssetIdsBySellable'
+import { filterBuyAssetsBySellAssetId } from 'lib/swapper/LifiSwapper/filterBuyAssetsBySellAssetId/filterBuyAssetsBySellAssetId'
+import { getTradeQuote } from 'lib/swapper/LifiSwapper/getTradeQuote/getTradeQuote'
+import { getUsdRate } from 'lib/swapper/LifiSwapper/getUsdRate/getUsdRate'
+import { SWAPPER_NAME, SWAPPER_TYPE } from 'lib/swapper/LifiSwapper/utils/constants'
+import { createLifiAssetMap } from 'lib/swapper/LifiSwapper/utils/createLifiAssetMap/createLifiAssetMap'
+import { createLifiChainMap } from 'lib/swapper/LifiSwapper/utils/createLifiChainMap/createLifiChainMap'
+import { getLifi } from 'lib/swapper/LifiSwapper/utils/getLifi'
+
+export class LifiSwapper implements Swapper<EvmChainId> {
+  readonly name = SWAPPER_NAME
+  private lifiChainMap: Map<ChainId, LifiChainKey> = new Map()
+  private lifiAssetMap: Map<AssetId, LifiToken> = new Map()
+  private lifiBridges: LifiBridgeDefinition[] = []
+
+  /** perform any necessary async initialization */
+  async initialize(): Promise<void> {
+    const supportedChainRefs = evmChainIds.map(
+      chainId => Number(fromChainId(chainId).chainReference) as LifiChainId,
+    )
+
+    const { bridges, chains, tokens } = await getLifi().getPossibilities({
+      include: ['bridges', 'chains', 'tokens'],
+      chains: supportedChainRefs,
+    })
+
+    if (chains !== undefined) this.lifiChainMap = createLifiChainMap(chains)
+    if (tokens !== undefined) this.lifiAssetMap = createLifiAssetMap(tokens)
+    if (bridges !== undefined) this.lifiBridges = bridges
+  }
+
+  /** Returns the swapper type */
+  getType(): SwapperType {
+    return SWAPPER_TYPE
+  }
+
+  /**
+   * Builds a trade with definitive rate & txData that can be executed with executeTrade
+   **/
+  async buildTrade(_args: BuildTradeInput): Promise<Trade<EvmChainId>> {
+    return await Promise.reject(new SwapError('LifiSwapper: buildTrade unimplemented'))
+  }
+
+  /**
+   * Get a trade quote
+   */
+  async getTradeQuote(input: GetEvmTradeQuoteInput): Promise<TradeQuote<EvmChainId>> {
+    return await getTradeQuote(input, this.lifiAssetMap, this.lifiChainMap, this.lifiBridges)
+  }
+
+  /**
+   * Get the usd rate from either the assets symbol or tokenId
+   */
+  async getUsdRate(asset: Asset): Promise<string> {
+    return await getUsdRate(asset, this.lifiChainMap)
+  }
+
+  /**
+   * Execute a trade built with buildTrade by signing and broadcasting
+   */
+  async executeTrade(_args: ExecuteTradeInput<ChainId>): Promise<TradeResult> {
+    return await Promise.reject(new SwapError('LifiSwapper: executeTrade unimplemented'))
+  }
+
+  /**
+   * Get a boolean if a quote needs approval
+   */
+  async approvalNeeded(input: ApprovalNeededInput<EvmChainId>): Promise<ApprovalNeededOutput> {
+    return await approvalNeeded(input)
+  }
+
+  /**
+   * Get the txid of an approve infinite transaction
```

Comment:
We might want to be explicit that not only does it *gets* a Txid, but it actually broadcasts the transaction
```suggestion
   * Broadcasts an infinite approval Tx and returns the Txid
```